memory/table.go

```diff
@@ -391,6 +391,16 @@ func (t tableEditor) Close(*sql.Context) error {
 	return nil
 }
 
+func (t tableEditor) Savepoint(ctx *sql.Context) {}
```

Comment:
If these aren't necessary, it's a sign that we're missing engine tests demonstrating why they are (very likely), or that the memory implementation doesn't correctly enforce constraints (this is for sure true)